### ISSUE TITLE
[Menu] Missing focus on the `menu button` when closing `submenu` with Escape key

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#2959]
### Provide a general summary of the issue here

When we have a `menu` with `submenu`, after closing the `submenu` with **Escape** key, focus is not going back to the `menu button`. When we have a `menu` without `submenu`, it works fine.

Please check those examples:
- With submenu: https://storybook.sanomalearning.design/?path=/story/actions-menu-menu-button--submenu (doesn't work properly)
- Without submenu: https://storybook.sanomalearning.design/?path=/story/actions-menu-menu-button--basic (works fine)

Please also check this comment from @KatarzynaWM :
https://github.com/sl-design-system/components/issues/2748#issuecomment-3804350536

### 👤 Your name

SLDS Team

### 🧢 Your Product/Team

SLDS
